cpp/src/spatial/pairwise_point_in_polygon.cu

```diff
@@ -137,13 +137,7 @@ std::unique_ptr<cudf::column> pairwise_point_in_polygon(cudf::column_view const&
   CUSPATIAL_EXPECTS(not poly_points_x.has_nulls() && not poly_points_y.has_nulls(),
                     "Polygon points must not contain nulls");
 
-  CUSPATIAL_EXPECTS(poly_ring_offsets.size() >= poly_offsets.size(),
-                    "Each polygon must have at least one ring");
-
-  CUSPATIAL_EXPECTS(poly_points_x.size() >= poly_offsets.size() * 4,
-                    "Each ring must have at least four vertices");
-
-  CUSPATIAL_EXPECTS(test_points_x.size() == poly_offsets.size(),
+  CUSPATIAL_EXPECTS(test_points_x.size() == std::max(poly_offsets.size() - 1, 0),
```

Comment:
This needs to pass when an empty test_points column is passed, regardless of whether the polygon offsets are malformed. If an empty offsets array is also passed, this will be `0 == -1`, and this will throw. Hence the need for std::max. Even though an empty polygon offsets column is not technically correct, I think we should consider it to mean "zero polygons".

Reply:
This needs to pass when an empty test_points column is passed, regardless of whether the polygon offsets are malformed. If an empty offsets array is also passed, this will be 0 == -1, and this will throw. Hence the need for std::max. Even though an empty polygon offsets column is not technically correct, I think we should consider it to mean "zero polygons".

Reply:
Ok... Note that in header only API you added the test macro, if the offset is malformed the API will throw anyway.

Reply:
If I remove `std::max` this test fails:
https://github.com/rapidsai/cuspatial/blob/e9a84b97000ddca42948bc89dad6ef231f522e70/cpp/tests/spatial/pairwise_point_in_polygon_test.cpp#L43

 If I leave it in, it passes.  I think we want this case to not throw.